### ISSUE TITLE
cannnot upgrae stack using `stack upgrade` 

### ISSUE THREAD

**Issue by matsubara0507** [commercialhaskell/stack#3175]
use Docker image [fpco/stack-build](https://hub.docker.com/r/fpco/stack-build/)

```
root@ed4ec9719bb3:/# stack --version
Version 1.3.2, Git revision 3f675146590da4f3edf768b89355f798229da2a5 (4395 commits) x86_64 hpack-0.15.0
root@ed4ec9719bb3:/# stack upgrade -v
Version 1.3.2, Git revision 3f675146590da4f3edf768b89355f798229da2a5 (4395 commits) x86_64 hpack-0.15.0
2017-05-15 01:56:35.784316: [info] Current Stack version: 1.3.2, available download version: 1.4.0
@(Stack/Upgrade.hs:151:17)
2017-05-15 01:56:35.785249: [info] Newer version detected, downloading
@(Stack/Upgrade.hs:167:13)
2017-05-15 01:56:35.785697: [info] Querying for archive location for platform: linux-x86_64-static
@(Stack/Setup.hs:1709:13)
2017-05-15 01:56:35.786228: [info] Downloading from: https://github.com/commercialhaskell/stack/releases/download/v1.4.0/stack-1.4.0-linux-x86_64-static.tar.gz
@(Stack/Setup.hs:1725:5)
2017-05-15 01:56:43.877640: [info] Download complete, testing executable
@(Stack/Setup.hs:1734:5)
Version 1.4.0, Git revision e714f1dd3fade19496d91bd6a017e435a96a6bcd (4640 commits) x86_64 hpack-0.17.0
2017-05-15 01:56:43.899100: [info] New stack executable available at /root/.local/bin/stack
@(Stack/Setup.hs:1753:5)
root@ed4ec9719bb3:/# stack --version
Version 1.3.2, Git revision 3f675146590da4f3edf768b89355f798229da2a5 (4395 commits) x86_64 hpack-0.15.0
```

**Comment by mgsloan** [comment]
Sorry for the delay in response.

Is `/root/.local/bin/` on your PATH?  In other words, what does `which stack` yield?

I'd expect stack to warn in this case, but the logic wasn't there for the binary upgrade path, so I've added it in https://github.com/commercialhaskell/stack/commit/7c08f523e1ea32922866b88aa1253277a9f97930

**Comment by matsubara0507** [comment]
This problem has been resolved by restart shell.

```
$ docker run --rm -it fpco/stack-build /bin/bash
root@48cf2979f01b:/# stack --version
Version 1.3.2, Git revision 3f675146590da4f3edf768b89355f798229da2a5 (4395 commits) x86_64 hpack-0.15.0
root@48cf2979f01b:/# echo $PATH
/root/.cabal/bin:/root/.local/bin:/opt/ghc/8.0.2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
root@48cf2979f01b:/# which stack
/usr/local/bin/stack
root@48cf2979f01b:/# /root/.local/bin/stack --version
bash: /root/.local/bin/stack: No such file or directory
root@48cf2979f01b:/# stack upgrade
Current Stack version: 1.3.2, available download version: 1.4.0
Newer version detected, downloading
Querying for archive location for platform: linux-x86_64-static
Downloading from: https://github.com/commercialhaskell/stack/releases/download/v1.4.0/stack-1.4.0-linux-x86_64-static.tar.gz
Download complete, testing executable
Version 1.4.0, Git revision e714f1dd3fade19496d91bd6a017e435a96a6bcd (4640 commits) x86_64 hpack-0.17.0
New stack executable available at /root/.local/bin/stack
root@48cf2979f01b:/# stack --version
Version 1.3.2, Git revision 3f675146590da4f3edf768b89355f798229da2a5 (4395 commits) x86_64 hpack-0.15.0
root@48cf2979f01b:/# echo $PATH
/root/.cabal/bin:/root/.local/bin:/opt/ghc/8.0.2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
root@48cf2979f01b:/# which stack
/root/.local/bin/stack
root@48cf2979f01b:/# exec $SHELL -l
root@48cf2979f01b:/# stack --version
Version 1.4.0, Git revision e714f1dd3fade19496d91bd6a017e435a96a6bcd (4640 commits) x86_64 hpack-0.17.0
root@48cf2979f01b:/# which stack
/root/.local/bin/stack
root@48cf2979f01b:/# echo $PATH
/root/.cabal/bin:/root/.local/bin:/opt/ghc/8.0.2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

PATH contained `/root/.local/bin/`  from the beginning.
But, when `/root/.local/bin/stack` is not exist at start of docker,  shell can not found new stack without restart.

Thanks to your advice, I was able to find the cause.
